### PR TITLE
A few documentation improvements for `syntax::print::pp`

### PR DESCRIPTION
**Pull request by bjorn3**
* Moved algorithm explanation to module docs
* Added ``` before and after the examples
* Explanation of the `rbox`, `ibox` and `cbox` names
* Added docs about the breaking types to `Breaks`

### CODE REVIEW

File: src/libsyntax/print/pp.rs [hunk 1]
```diff
@@ -60,11 +60,119 @@
...
+///     good, friends);
+/// ```
+///
+/// whereas a consistent breaking would yield:
+///
+/// ```
+/// foo(hello,
+///     there
+///     good,
+///     friends);
```
Comment (jseyfried) [inline src/libsyntax/print/pp.rs]: This looks like a duplicate of the comment starting [here](https://github.com/rust-lang/rust/pull/39557/files#diff-d774485776497d058e42e27b18d536f2R79). I would just give a one line summary here and say it's described in more detail in the module docs.

**Review by jseyfried — APPROVED** [review]
r=me with commits squahsed

**Comment by bjorn3** [comment]
Squashed

**Pull request merged**